internal/gatewayapi/translator.go

```diff
@@ -1175,18 +1176,37 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 							)
 						}
 					case v1beta1.HTTPRouteFilterExtensionRef:
-						// "If a reference to a custom filter type cannot be resolved, the filter MUST NOT be skipped.
-						// Instead, requests that would have been processed by that filter MUST receive a HTTP error response."
-						errMsg := fmt.Sprintf("Unknown custom filter type: %s", filter.Type)
-						parentRef.SetCondition(httpRoute,
-							v1beta1.RouteConditionAccepted,
-							metav1.ConditionFalse,
-							v1beta1.RouteReasonUnsupportedValue,
-							errMsg,
-						)
-						directResponse = &ir.DirectResponse{
-							Body:       &errMsg,
-							StatusCode: 500,
+						// If a reference to a custom filter type cannot be resolved, the filter MUST NOT be skipped.
+						// Instead, requests that would have been processed by that filter MUST receive an HTTP error
+						// response.
+						valid := false
+						found := false
+						errMsg := fmt.Sprintf("Unsupported filter type: %s", filter.Type)
+						if err := ValidateHTTPRouteFilter(&filter); err != nil {
+							errMsg = fmt.Sprintf("Invalid filter: %v", err)
+						} else {
+							valid = true
+							for _, f := range resources.AuthenFilters {
+								if f.Namespace == h.Namespace && f.Name == string(filter.ExtensionRef.Name) {
+									found = true
+									break
+								}
+							}
+						}
+						if !found && valid {
+							errMsg = fmt.Sprintf("Reference not found for filter type: %s", filter.Type)
+						}
+						if !valid || !found {
+							parentRef.SetCondition(httpRoute,
+								v1beta1.RouteConditionAccepted,
+								metav1.ConditionFalse,
+								v1beta1.RouteReasonUnsupportedValue,
```

Comment:
It's not 100% clear in the Gateway API spec, but I think if the filter ref can't be resolved (i.e. the referent doesn't exist), then setting a `ResolvedRefs: false` condition would be more appropriate than `Accepted: false (UnsupportedValue)`.

Reply:
(For unknown filter types, what we're doing here still makes sense though)

Reply:
@skriss thanks for the review. Commit 07ec1b4 resolves your above comment.